src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -258,17 +288,80 @@ private void EnsureGlobbingUrlBuilder()
             }
         }
 
-        private static void BuildLinkTag(IDictionary<string, string> attributes, TagHelperContent builder)
+        private void BuildLinkTag(IDictionary<string, string> attributes, TagHelperContent builder)
         {
             builder.Append("<link ");
 
             foreach (var attribute in attributes)
             {
-                builder.Append(
-                    string.Format(CultureInfo.InvariantCulture, "{0}=\"{1}\" ", attribute.Key, attribute.Value));
+                if (attribute.Key == "href")
+                {
+                    AppendHref(builder, attribute.Key, attribute.Value);
+                }
+                else
+                {
+                    builder.Append(
+                        string.Format(CultureInfo.InvariantCulture, "{0}=\"{1}\" ", attribute.Key, attribute.Value));
+                }
             }
 
             builder.Append("/>");
         }
+
+        private void AppendHref(TagHelperContent content, string hrefKey, string hrefValue)
+        {
+            string formatter = "{0}=\"{1}";
+            string fileVersion = string.Empty;
+            if (FileVersion == "true")
+            {
+                fileVersion = GetFileContentEncodedString(hrefValue);
+                // This is the case if there was any exception thrown when reading the file.
+                if (fileVersion != string.Empty)
+                {
+                    formatter += "?v={2}";
+                }
+            }
+
+            formatter += "\" ";
+
+            content.Append(
+                string.Format(
+                    CultureInfo.InvariantCulture,
+                    formatter,
+                    hrefKey,
+                    HtmlEncoder.HtmlEncode(hrefValue),
```

Comment:
Why were you encoding before? We don't for other attributes (updated my snippet that takes care of the `hrefValue` above)